File: SDKs/googleapis.com/cloudasset/v1/typescript/src/sdk/models/shared/analyzerorgpolicyconstraint.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { GoogleCloudAssetV1CustomConstraint } from "./googlecloudassetv1customconstraint";
import { GoogleCloudAssetV1Constraint } from "./googlecloudassetv1constraint";



// AnalyzerOrgPolicyConstraint
/** 
 * The organization policy constraint definition.
**/
export class AnalyzerOrgPolicyConstraint extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=customConstraint" })
  customConstraint?: GoogleCloudAssetV1CustomConstraint;

  @SpeakeasyMetadata({ data: "json, name=googleDefinedConstraint" })
  googleDefinedConstraint?: GoogleCloudAssetV1Constraint;
}

File: SDKs/googleapis.com/cloudasset/v1/python/src/sdk/models/shared/analyzerorgpolicyconstraint.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class AnalyzerOrgPolicyConstraint:
    r"""AnalyzerOrgPolicyConstraint
    The organization policy constraint definition.
    """
    
    custom_constraint: Optional[GoogleCloudAssetV1CustomConstraint] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('customConstraint') }})
    google_defined_constraint: Optional[GoogleCloudAssetV1Constraint] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('googleDefinedConstraint') }})
    
